src/drivers/esc_rpm/esc_rpm.cpp

```diff
@@ -0,0 +1,242 @@
+/****************************************************************************
+ *
+ *   Copyright (c) 2019 PX4 Development Team. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name PX4 nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#include "esc_rpm.h"
+
+int
+EscRpm::task_spawn(int argc, char *argv[])
+{
+	auto *escRpm = new EscRpm();
+
+	if (!escRpm) {
+		PX4_ERR("driver allocation failed");
+		return PX4_ERROR;
+	}
+
+	_object.store(escRpm);
+	_task_id = task_id_is_work_queue;
+
+	escRpm->start();
+
+	return PX4_OK;
+}
+
+EscRpm::EscRpm() :
+	ScheduledWorkItem(MODULE_NAME, px4::wq_configurations::lp_default)
+{
+	param_get(param_find("ESC_RPM_PULSES"), &_pulses_per_revolution);
+
+	for (size_t i = 0; i < MAX_MOTORS; i++) {
+		_motor_data_array[i].timed_out = true;
+	}
+}
+
+void
+EscRpm::start()
+{
+	// Initialize the GPIO rising edge ISRs and callback
+	gpio_init();
+
+	ScheduleOnInterval(100_ms);
+}
+
+void
+EscRpm::gpio_init()
+{
+	irqstate_t state = px4_enter_critical_section();
+
+	/* * * * * * * * * * * * * * * * * *
+	*
+	* Add more GPIO inputs here and define them in board_config.h if you'd like more.
+	*
+	* GPIO_ESC_RPM_1 = PWM CH5
+	* GPIO_ESC_RPM_2 = PWM CH6
+	*
+	*/
+
+	// Configure GPIO
+	px4_arch_configgpio(GPIO_ESC_RPM_1);
+	px4_arch_configgpio(GPIO_ESC_RPM_2);
```

Comment:
Someday we may have a nice pin config interface like what Beat put together for I2C and SPI...